src/main/java/Hachi.java

```diff
@@ -0,0 +1,266 @@
+     * <p>3. say goodbye to the user with user input "bye" or "goodbye"
+     * <p>4. add a to-do to the list of task with "todo <event name>"
+     * <p>5. add a deadline to the list of task with "deadline <event name> /by <by date>"
+     * <p>6. add an event to the list of task with "event <event name> /from <start date> /to <end date>"
+     * <p>7. retrieve a list of chatbot commands with "help"
+     *
+     * @param args Command line arguments - not used.
+     */
+
+    public static void main(String[] args) {
+        spacerInsert("medium", false);
+        boolean isBye = false;
+        Task[] listOfTasks = new Task[100];
+        printGreetingMessage();
+        printHelpMessage();
+
+        while (!isBye) {
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+            String cleanedInput = line.toUpperCase().trim();
+            int indexOfSpace = cleanedInput.indexOf(" ");
+            String firstWord;
+            if (indexOfSpace == -1) { // check for single-word inputs
+                firstWord = cleanedInput;
+            } else {
+                firstWord = cleanedInput.substring(0, indexOfSpace);
+            }
+
+            if (firstWord.contains("MARK")) {
+                int indexOfTask = cleanedInput.indexOf("MARK") + 4; // find index of task number
+                int taskNumber = Integer.parseInt(cleanedInput.substring(indexOfTask).trim()); // parse string to int
+                if (cleanedInput.contains("UNMARK")) {
+                    unmarkTask(taskNumber - 1, listOfTasks);
+                } else {
+                    markTask(taskNumber - 1, listOfTasks);
+                }
+            } else {
+                switch (firstWord) {
+                case "LIST":
+                    retrieveList(listOfTasks);
+                    break;
+
+                case "TODO":
+                case "EVENT":
+                case "DEADLINE":
+                    TaskType currentTask;
+                    if (cleanedInput.startsWith("TODO")) {
+                        currentTask = TaskType.TODO;
+                    } else if (cleanedInput.startsWith("DEADLINE")) {
+                        currentTask = TaskType.DEADLINE;
+                    } else {
+                        currentTask = TaskType.EVENT;
+                    }
+                    addTask(currentTask, listOfTasks, line, cleanedInput);
+                    break;
+
+                case "BYE":
+                case "GOODBYE":
+                    isBye = true;
+                    printGoodbyeMessage();
+                    break;
+
+                case "HELP":
+                    printHelpMessage();
+                    break;
+
+                default:
+                    // handle invalid case
+                    System.out.println("Invalid command read."); // for testing
+                    break;
+                }
+            }
+        }
+    }
+}
```

Comment:
Good job overall on having a relatively concise and understandable code! Your documentation also allowed it to be understood easily as well!